### PR TITLE
ranking: Optimize reduce query

### PR DESCRIPTION
**Pull request by efritz**
Created an index and changed the join condition so that the name lookup doesn't require a hash of all repo names 😬.

## Test plan

Unit tests still pass.

### CODE REVIEW

File: enterprise/internal/codeintel/uploads/internal/store/store_ranking.go [hunk 1]
```diff
@@ -290,17 +290,21 @@ func (s *store) InsertPathRanks(
...
-		(SELECT id FROM repo WHERE name = repository) AS repository_id,
-		document_path AS path,
-		count
-	FROM codeintel_ranking_path_counts_inputs
+		pci.id,
+		r.id AS repository_id,
+		pci.document_path AS path,
+		pci.count
+	FROM codeintel_ranking_path_counts_inputs pci
+	JOIN repo r ON lower(r.name) = (lower(pci.repository::text) COLLATE "C")
```
Comment (efritz) [inline enterprise/internal/codeintel/uploads/internal/store/store_ranking.go:301]: I plan to follow-up to this to see if we can replace this column entirely with ids so we don't need a cross-lookup again.

File: migrations/frontend/1677594756_add_better_index_to_ranking_path_counts_table/up.sql [hunk 2]
```diff
@@ -0,0 +1,4 @@
+CREATE INDEX CONCURRENTLY IF NOT EXISTS codeintel_ranking_path_counts_inputs_graph_key_repository_id_processed
```
Comment (efritz) [inline migrations/frontend/1677594756_add_better_index_to_ranking_path_counts_table/up.sql:1]: Note: already created on dotcom

File: internal/database/schema.md [hunk 3]
```diff
@@ -1002,6 +1002,7 @@ Foreign-key constraints:
 Indexes:
     "codeintel_ranking_path_counts_inputs_pkey" PRIMARY KEY, btree (id)
     "codeintel_ranking_path_counts_inputs_graph_key_and_repository" btree (graph_key, repository)
+    "codeintel_ranking_path_counts_inputs_graph_key_repository_id_pr" btree (graph_key, repository, id) INCLUDE (document_path) WHERE NOT processed
```
Comment (Strum355) [inline internal/database/schema.md:1005]: Do we want to include `count` given its in the select?

Comment (efritz) [reply internal/database/schema.md:1005]: I would but I think this index is actually the non-dominating factor. It's already on dotcom and will likely be replaced when we do [this](https://github.com/sourcegraph/sourcegraph/pull/48361#discussion_r1120318292).

Comment (Strum355) [reply internal/database/schema.md:1005]: Why should we include `document_path` here then if we need to do a heap fetch anyway for `count`?

Comment (efritz) [reply internal/database/schema.md:1005]: Because I made the wrong index my dude.

Comment (Strum355) [reply internal/database/schema.md:1005]: so relatable

**Review by Strum355 — APPROVED** [review]

**Pull request merged**